test/e2e/data/infrastructure-docker/v1beta1/bases/cluster-with-topology.yaml

```diff
@@ -27,9 +27,11 @@ spec:
         replicas: ${WORKER_MACHINE_COUNT}
         failureDomain: fd4
     variables:
-    - name: etcdImageTag
       # We set an empty value to use the default tag kubeadm init is using.
+    - name: etcdImageTag
       value: ""
-    - name: coreDNSImageTag
       # We set an empty value to use the default tag kubeadm init is using.
+    - name: coreDNSImageTag
       value: ""
+    - name: kubeadmControlPlaneMaxSurge
```

Comment:
Can you give some context why that variable was added?

It doesn't seem to play well with kustomization on top, e.g. cluster-template-kcp-scale-in

Reply:
It was added to keep it in line with the current implementation - clusterclass-quick-start-runtime-sdk has a variable and patch for 

What's the issue with the scale in test? Isn't it alright as this is a different base?

Reply:
It uses this as base + adds the variable so we probably now have it twice

Reply:
```
make -C test/e2e cluster-templates
```

> test/e2e/data/infrastructure-docker/v1beta1/cluster-template-kcp-scale-in.yaml
```yaml
    - name: kubeadmControlPlaneMaxSurge
      value: "1"
    - name: kubeadmControlPlaneMaxSurge
      value: "0"
```

If we implemented it correctly this will fail on the validation with duplicate variable